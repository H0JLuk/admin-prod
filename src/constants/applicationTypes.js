const applicationTypes = [
    {type: 'IOS'},
    {type: 'ANDROID'},
    {type: 'OTHER'}
];

export default applicationTypes;