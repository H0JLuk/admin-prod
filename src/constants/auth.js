export const LOGIN_TYPE = {
    PASSWORD: 'PASSWORD',
    // SBOL_PRO: 'SBOL_PRO',
    SBER_USER_ID: 'SBER_USER_ID',
};