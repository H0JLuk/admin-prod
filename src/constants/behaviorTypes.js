const behaviorTypes = {
    QR: 'QrBehavior',
    OKKO: 'OkkoBehavior',
    SBER_LOGISTIC: 'SberLogisticBehavior',
    DUMMY: 'DummyBehavior',
    EMPTY: 'Empty', //technical behavior to display empty labels while awaiting server response
    WEB: 'WebOnlyBehavior',
};

export default behaviorTypes;
