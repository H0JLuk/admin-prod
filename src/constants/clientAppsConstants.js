export const APP_MECHANICS = {
    PRESENTS: { value: 'PRESENTS', label: 'Подарки' },
    ECOSYSTEM: { value: 'ECOSYSTEM', label: 'Продукты' },
    PRESENTATION: { value: 'PRESENTATION', label: 'Экскурсия' },
    BUNDLE: { value: 'BUNDLE', label: 'WOW' },
};

export const MECHANICS_CHECKBOXES = Object.keys(APP_MECHANICS).map(key => ({
    value: APP_MECHANICS[key].value,
    label: APP_MECHANICS[key].label,
}));

export const APP_LOGIN_TYPES = [
    { label: 'Логин/Пароль', value: 'PASSWORD' },
    { label: 'СБОЛ ПРО', value: 'SBOL_PRO' },
    { label: 'Ссылка', value: 'DIRECT_LINK' },
];

export const MECHANICS_ERROR = `Необходимо выбрать хотя бы одну обязательную механику (${APP_MECHANICS.PRESENTS.label}, ${APP_MECHANICS.ECOSYSTEM.label} или ${APP_MECHANICS.BUNDLE.label})`;
