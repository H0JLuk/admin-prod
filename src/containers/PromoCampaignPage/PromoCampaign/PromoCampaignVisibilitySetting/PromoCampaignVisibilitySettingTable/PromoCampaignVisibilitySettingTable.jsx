import React from 'react';
import { Table, Switch } from 'antd';

import styles from './PromoCampaignVisibilitySettingTable.module.css';

const LOCATION = 'Локация';
const SALE_POINT = 'Точка продажи';

const ColumnRender = (item, name) => (
    <>
        <div className={ styles.title }>{ name }</div>
        <div className={ styles.item }>{ item }</div>
    </>
);

const PromoCampaignVisibilitySettingTable = ({
    rowSelection,
    loading,
    dataSource,
    onChangeVisible,
    selectRow,
    pagination,
    onChange,
}) =>  (
        <Table
            className={ styles.table }
            loading={ loading }
            rowKey="id"
            pagination={ pagination }
            rowSelection={ rowSelection }
            dataSource={ dataSource }
            showHeader={ false }
            onRow={ (record) => ({ onClick: () => selectRow(record.id) }) }
            onChange={ onChange }
        >
        <Table.Column
            width="40%"
            key="locationName"
            dataIndex="locationName"
            render = { (item) => ColumnRender(item, LOCATION) }
        />
        <Table.Column
            width="40%"
            key="salePointName"
            dataIndex="salePointName"
            render = { (item) => ColumnRender(item, SALE_POINT) }
        />
        <Table.Column
            width="20%"
            key="visible"
            className={ styles.switch }
            render={ (item) => <Switch checked={ item.visible } onClick={ () => onChangeVisible({ ...item, visible: !item.visible }) } /> }
        />
        </Table>
);

export default PromoCampaignVisibilitySettingTable;
