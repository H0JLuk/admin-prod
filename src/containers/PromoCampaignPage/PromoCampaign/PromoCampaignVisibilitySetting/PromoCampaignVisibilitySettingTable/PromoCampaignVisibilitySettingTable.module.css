.title {
    font-family: SBSansText-Medium;
    margin-top: 0;
    font-size: 16px;
    color: #657182;
}

.item {
    font-family: SBSansText-Medium;
    font-size: 16px;
    min-height: 22px;
    color: #171725;
}

.switch {
    text-align: right;
}

.table :global(.ant-table) {
    font-size: 16px;
    line-height: 24px;
    letter-spacing: 0.18px;
}

.table :global(.ant-table-tbody > tr > td) {
    border-bottom: none;
    padding: 15px;
}

.table :global(.ant-table-tbody td:first-child) {
    padding-left: 25px;
}

.table :global(.ant-table-tbody td:last-child) {
    padding-right: 25px;
}

.table :global(.ant-checkbox-checked:after),
.table :global(.ant-checkbox-checked .ant-checkbox-inner) {
    border: none;
}

.table :global(.ant-table) {
    border-radius: 5px;
}

.table :global(.ant-table-tbody > tr:hover) {
    cursor: pointer;
}

.table :global(.ant-select-focused:not(.ant-select-disabled).ant-select-single .ant-select-selector),
.table :global(.ant-pagination-options-quick-jumper input:focus) {
    box-shadow: none !important;
}
