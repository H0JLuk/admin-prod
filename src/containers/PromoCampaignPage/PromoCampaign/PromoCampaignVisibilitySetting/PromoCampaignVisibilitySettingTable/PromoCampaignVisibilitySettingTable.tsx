import React from 'react';
import { Table, Switch, TablePaginationConfig, TableProps } from 'antd';
import { VisibilitySettingDto } from '@types';

import styles from './PromoCampaignVisibilitySettingTable.module.css';

const LOCATION = 'Локация';
const SALE_POINT = 'Точка продажи';

const ColumnRender = (item: string, name: string) => (
    <>
        <div className={styles.title}>{name}</div>
        <div className={styles.item}>{item}</div>
    </>
);

type PromoCampaignVisibilitySettingTableProps = {
    rowSelection: TableProps<VisibilitySettingDto>['rowSelection'];
    loading: boolean;
    dataSource: VisibilitySettingDto[];
    onChangeVisible: (val: VisibilitySettingDto) => void;
    selectRow: (id: number) => void;
    pagination: TablePaginationConfig;
    onChange: TableProps<VisibilitySettingDto>['onChange'];
};

const PromoCampaignVisibilitySettingTable: React.FC<PromoCampaignVisibilitySettingTableProps> = ({
    rowSelection,
    loading,
    dataSource,
    onChangeVisible,
    selectRow,
    pagination,
    onChange,
}) => (
    <Table<VisibilitySettingDto>
        className={styles.table}
        loading={loading}
        rowKey="id"
        pagination={pagination}
        rowSelection={rowSelection}
        dataSource={dataSource}
        showHeader={false}
        onRow={(record) => ({ onClick: () => selectRow(record.id) })}
        onChange={onChange}
    >
        <Table.Column
            width="40%"
            key="locationName"
            dataIndex="locationName"
            render={(item) => ColumnRender(item, LOCATION)}
        />
        <Table.Column
            width="40%"
            key="salePointName"
            dataIndex="salePointName"
            render={(item) => ColumnRender(item, SALE_POINT)}
        />
        <Table.Column
            width="20%"
            key="visible"
            className={styles.switch}
            render={(item) => (
                <Switch
                    checked={item.visible}
                    onClick={(_, e) => {
                        e.stopPropagation();
                        onChangeVisible({ ...item, visible: !item.visible });
                    }}
                />
            )}
        />
    </Table>
);

export default PromoCampaignVisibilitySettingTable;
