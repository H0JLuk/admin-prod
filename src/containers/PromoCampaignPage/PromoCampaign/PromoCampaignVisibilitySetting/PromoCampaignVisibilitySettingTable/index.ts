import PromoCampaignVisibilitySettingTable from './PromoCampaignVisibilitySettingTable';

export default PromoCampaignVisibilitySettingTable;
