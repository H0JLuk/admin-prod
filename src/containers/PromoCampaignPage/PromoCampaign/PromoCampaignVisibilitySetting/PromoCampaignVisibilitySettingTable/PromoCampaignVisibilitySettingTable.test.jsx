import React from 'react';
import { shallow } from 'enzyme';

import PromoCampaignVisibilitySettingTable from './PromoCampaignVisibilitySettingTable';

import { visibilitySettingsTestArray } from '../../../../../../__tests__/constants';


const promoCampaignVisibilitySettingTablePropsTestData = {
    loading: false,
    dataSource: visibilitySettingsTestArray,
    pagination: {
        current: 1,
        total: 2,
        pageSize: '10',
        locale: {
            items_per_page: '',
            prev_page: 'Назад',
            next_page: 'Вперед',
            jump_to: 'Перейти',
            prev_5: 'Предыдущие 5',
            next_5: 'Следующие 5',
            prev_3: 'Предыдущие 3',
            next_3: 'Следующие 3'
        },
        showSizeChanger: true,
        showQuickJumper: true
    },
    rowSelection: undefined,
    selectedRow: '1',
    onChange() {
        return;
    },
    onChangeVisible() {
        return;
    }
};


describe('<PromoCampaignVisibilitySettingTable /> tests', () => {
    const props = {
        ...promoCampaignVisibilitySettingTablePropsTestData,
    };

    it('PromoCampaignVisibilitySettingTable snapshot', () => {
        const wrapper = shallow(<PromoCampaignVisibilitySettingTable { ...props } />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
