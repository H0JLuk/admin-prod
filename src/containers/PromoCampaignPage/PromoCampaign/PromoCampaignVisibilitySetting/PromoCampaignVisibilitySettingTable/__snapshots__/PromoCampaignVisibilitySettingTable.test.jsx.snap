// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoCampaignVisibilitySettingTable /> tests PromoCampaignVisibilitySettingTable snapshot 1`] = `
"<Table className=\\"table\\" loading={false} rowKey=\\"id\\" pagination={{...}} rowSelection={[undefined]} dataSource={{...}} showHeader={false} onRow={[Function: onRow]} onChange={[Function: onChange]}>
  <Column width=\\"40%\\" dataIndex=\\"locationName\\" render={[Function: render]} />
  <Column width=\\"40%\\" dataIndex=\\"salePointName\\" render={[Function: render]} />
  <Column width=\\"20%\\" className=\\"switch\\" render={[Function: render]} />
</Table>"
`;
